Marlin/src/module/stepper.cpp

```diff
@@ -1684,11 +1688,17 @@ void Stepper::pulse_phase_isr() {
 
     #define PULSE_PREP_SHAPING(AXIS, DIVIDEND) do{ \
       delta_error[_AXIS(AXIS)] += (DIVIDEND); \
-      if ((MAXDIR(AXIS) && delta_error[_AXIS(AXIS)] <= -0x30000000L) || (MINDIR(AXIS) && delta_error[_AXIS(AXIS)] >= 0x30000000L)) { \
+      if ((MAXDIR(AXIS) && delta_error[_AXIS(AXIS)] < -0x30000000L) || (MINDIR(AXIS) && delta_error[_AXIS(AXIS)] > 0x30000000L)) { \
         TBI(last_direction_bits, _AXIS(AXIS)); \
```

Comment:
Could `last_direction_bits` change after the block regular pulses but before the shaping pulses have completed? If so, this would seem to cause the final shaping pulses to be applied in the wrong direction.

Reply:
Yes, in some cases, it could. This would be the correct behaviour in those cases. What will have happened is that IS caused the corner to be cut so that the segment before the direction change does not drive the axis all the way. The beginning of one segment and the end of the other get averaged together.